### ISSUE TITLE
feat(train_test_split): Refine detection of ML task

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#1080]
### Is your feature request related to a problem? Please describe.

@koaning got the following error when running `skore.train_test_split`:
> /workspaces/joblib-modal/.venv/lib/python3.12/site-packages/skore/sklearn/train_test_split/train_test_split.py:165: HighClassImbalanceTooFewExamplesWarning: It seems that you have a classification problem with at least one class with fewer than 100 examples in the test set. In this case, using train_test_split may not be a good idea because of high variability in the scores obtained on the test set. We suggest three options to tackle this challenge: you can increase test_size, collect more data, or use skore's cross_validate function.

and wonders why skore is assuming that he is doing classification, while he is actually doing a regression.

@augustebaum points out that if the target vector contains a lot of integers, scikit-learn still interprets it as a classification problem.

### Describe the solution you'd like

If there are many integers in the target of a regression task, it should not be detected as a classification task.

@koaning suggests the following ideas:
- what about checking if the ints are sequential?
- And if it also includes a zero class?
